src/Tolerance/Bridge/Guzzle/Operation/RetryEvaluator/PromiseRetryEvaluator.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+/*
+ * This file is part of the Tolerance package.
+ *
+ * (c) Samuel ROZE <samuel.roze@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Tolerance\Bridge\Guzzle\Operation\RetryEvaluator;
+
+use GuzzleHttp\Psr7\Response;
+use Tolerance\Operation\RetryEvaluator\RetryEvaluator;
+
+class PromiseRetryEvaluator implements RetryEvaluator
```

Comment:
Following our discussions, can you then replace the `RetryEvaluator`s by the existing [`ThrowableCatcherVoter`](https://github.com/Tolerance/Tolerance/blob/0edff0776580c948562a9de7b5fbe40ec0dae628/src/Tolerance/Operation/ExceptionCatcher/ThrowableCatcherVoter.php#L22) ? 

Also, in order to have a clearer name, could you replace the class name by something like `RequestServerErrorVoter` ?